### ISSUE TITLE
[Bug]: elpaca-use-package Requires use-package minimum version: 2.4.4

### ISSUE THREAD

**Issue by purplg** [progfolio/elpaca#237]
### Confirmation

- [X] I have checked the documentation (README, Wiki, docstrings, etc)
- [ ] I am  checking these without reading them.
- [X] I have searched previous issues to see if my question is a duplicate.

### Elpaca Version

Elpaca 64e2ef4 HEAD -> master, origin/master, origin/HEAD
installer:      0.6
emacs-version:  GNU Emacs 29.1.90 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.38, cairo version 1.18.0)
 of 2023-12-14
git --version:  git version 2.42.0

### Operating System

ArchLinux

### Description

I am now getting the error in the title on the initial **elpaca-log** buffer when it's trying to install **elpaca-use-package**.

I am able to get **elpaca-use-package** to load by modifying its' use-package version in `Package-Requires` to `0.0.0`.

When doing this, the variable `use-package-version` shows `2.4.5` on my system.

This error occurs even when using the following minimal config shown here:

```emacs-lisp
;;; --- -*- lexical-binding: t; -*-
(defvar elpaca-installer-version 0.6)
(defvar elpaca-directory (expand-file-name "elpaca/" user-emacs-directory))
(defvar elpaca-builds-directory (expand-file-name "builds/" elpaca-directory))
(defvar elpaca-repos-directory (expand-file-name "repos/" elpaca-directory))
(defvar elpaca-order '(elpaca :repo "https://github.com/progfolio/elpaca.git"
                              :ref nil
                              :files (:defaults "elpaca-test.el" (:exclude "extensions"))
                              :build (:not elpaca--activate-package)))
(let* ((repo  (expand-file-name "elpaca/" elpaca-repos-directory))
       (build (expand-file-name "elpaca/" elpaca-builds-directory))
       (order (cdr elpaca-order))
       (default-directory repo))
  (add-to-list 'load-path (if (file-exists-p build) build repo))
  (unless (file-exists-p repo)
    (make-directory repo t)
    (when (< emacs-major-version 28) (require 'subr-x))
    (condition-case-unless-debug err
        (if-let ((buffer (pop-to-buffer-same-window "*elpaca-bootstrap*"))
                 ((zerop (call-process "git" nil buffer t "clone"
                                       (plist-get order :repo) repo)))
                 ((zerop (call-process "git" nil buffer t "checkout"
                                       (or (plist-get order :ref) "--"))))
                 (emacs (concat invocation-directory invocation-name))
                 ((zerop (call-process emacs nil buffer nil "-Q" "-L" "." "--batch"
                                       "--eval" "(byte-recompile-directory \".\" 0 'force)")))
                 ((require 'elpaca))
                 ((elpaca-generate-autoloads "elpaca" repo)))
            (progn (message "%s" (buffer-string)) (kill-buffer buffer))
          (error "%s" (with-current-buffer buffer (buffer-string))))
      ((error) (warn "%s" err) (delete-directory repo 'recursive))))
  (unless (require 'elpaca-autoloads nil t)
    (require 'elpaca)
    (elpaca-generate-autoloads "elpaca" repo)
    (load "./elpaca-autoloads")))
(add-hook 'after-init-hook #'elpaca-process-queues)
(elpaca `(,@elpaca-order))

(elpaca elpaca-use-package
  (elpaca-use-package-mode)
  (setq elpaca-use-package-by-default t))
(elpaca-wait)
```

**Emacs commit**:`8729a2a10d9b8d88f6ba33b5ce62f74d89e7788a`
**Configured with**: `./configure --with-imagemagick --with-json --with-tree-sitter --with-pgtk --with-native-compilation=aot`

**Comment by progfolio** [comment]
Thanks for taking the time to fill out a report.
Could you please share the value of the `package--builtin-versions` variable?

**Comment by purplg** [comment]
Sure thing:

```
((xref 1 6 . #12=(3))
 (whitespace 13 . #5=(2 . #1=(2)))
 (viper 3 14 . #1#)
 (verilog-mode 2021 10 14 127365406)
 (vera-mode 2 . #7=(28))
 (use-package . #6=(0 . #1#))
 (use-package 2 4 . #9=(5))
 (tramp 2 6 2 29 . #1#)
 (tildify 4 6 . #3=(1))
 (thunk . #2=(1 . #14=(0)))
 (tabulated-list . #2#)
 (svg 1 . #3#)
 (srecode . #4=(1 . #1#))
 (sql 3 . #8=(6))
 (soap-client 3 2 . #3#)
 (so-long 1 . #4#)
 (seq 2 24)
 (semantic . #5#)
 (ruby-ts-mode . #6#)
 (ruby-mode . #4#)
 (python 0 . #7#)
 (pulse . #2#)
 (ps-mode 1 1 . #10=(9))
 (project 0 9 8)
 (package 1 . #2#)
 (org 9 6 15)
 (ntlm 2 . #2#)
 (nadvice . #2#)
 (mixal-mode 0 . #15=(4))
 (mh-e 8 6 -4)
 (meta-mode . #2#)
 (map 3 3 . #3#)
 (let-alist 1 . #13=(0 . #8#))
 (jsonrpc 1 0 16)
 (json . #11=(1 . #9#))
 (js . #10#)
 (isearchb . #11#)
 (info-xref . #12#)
 (image-dired 0 . #9#)
 (idlwave 6 1 22)
 (htmlfontify 0 21)
 (gnus 5 13)
 (foldout 1 10)
 (flymake-proc . #2#)
 (flymake 1 . #5#)
 (find-cmd . #13#)
 (feedmail 11)
 (faceup 0 . #13#)
 (external-completion 0 . #3#)
 (eshell 2 . #19=(4 . #1#))
 (erc 5 5 0 29 . #3#)
 (epg 1 . #18=(0 . #14#))
 (eldoc 1 13 . #14#)
 (eieio-core . #16=(1 . #15#))
 (eieio . #16#)
 (eglot 1 12 29)
 (ediff 2 81 . #8#)
 (ede . #17=(2 . #14#))
 (cl-print . #2#)
 (cl-lib . #2#)
 (cl-generic . #2#)
 (cfengine . #16#)
 (cedet . #17#)
 (cc-mode 5 33 . #3#)
 (bind-key 2 4 . #3#)
 (backtrace . #2#)
 (auth-source-pass 5 . #18#)
 (antlr-mode 2 . #20=(2 . #12#))
 (ansi-color 3 . #19#)
 (allout-widgets . #2#)
 (allout . #20#)
 (emacs 29 1 90))
```

**Comment by progfolio** [comment]
Thank you.
Emacs provides a file, loaddefs.el, which populates the `package--builtin-versions` variable.
Elpaca uses that variable to determine the version of built-in packages as well.

I built Emacs from source from the commit you're on and was able to reproduce the same issue.
Note the duplicate use-package entry in `package--builtin-versions`:

```emacs-lisp
(use-package . #6=(0 . #1#))
(use-package 2 4 . #9=(5))
```
The erroneous version is added by use-package-ensure-system-package according to the loaddefs file my build of Emacs produced. This is an upstream Emacs bug (I've reported it). There are a few options:

If you're fine with updating Emacs, it looks like the loaddefs are corrected in emacs-29.0.91.
If you want to stay on your current version the following can be added your init file prior to the Elpaca installer:

``` emacs-lisp
;;TEMP WORKAROUND FOR Emacs 29.1.90's incorrect loaddefs
(setq package--builtin-versions (reverse package--builtin-versions))
```

Does that help?
